src/compiler/checker.ts

```diff
@@ -6688,6 +6684,10 @@ namespace ts {
                 }
                 // At this point we can bail if it's not a union

```

Comment:
This comment should be updated. It's only sort-of true now that there are up to two decisions made before returning.